src/khoj/configure.py

```diff
@@ -78,16 +79,20 @@ def configure_search_types(config: FullConfig):
     core_search_types = {e.name: e.value for e in SearchType}
     # Extract configured plugin search types
     plugin_search_types = {}
-    if config.content_type.plugins:
+    if config.content_type and config.content_type.plugins:
         plugin_search_types = {plugin_type: plugin_type for plugin_type in config.content_type.plugins.keys()}
 
     # Dynamically generate search type enum by merging core search types with configured plugin search types
     return Enum("SearchType", merge_dicts(core_search_types, plugin_search_types))
 
 
-def configure_search(model: SearchModels, config: FullConfig, regenerate: bool, t: state.SearchType = None):
+def configure_search(model: SearchModels, config: FullConfig, regenerate: bool, t: Optional[state.SearchType] = None):
+    if config.content_type is None or config.search_type is None:
+        logger.error("🚨 Content Type or Search Type not configured.")
```

Comment:
During first-run the content_type isn't expected to be populated. So putting a `pydantic` validator inside FullConfig would throw earlier than is desired. Here we're only saying we'll configure search if content_type is defined but we do not throw. So calls on startup during first_run will only log the error but not throw. 

Realized 2 things though:
1. Code should actually just log warning, instead of log error as no search or content type being configured is expected during first run
2. Code should check if `config is None` as welll